### PR TITLE
CSI: HTTP handlers for create/delete/list

### PR DESCRIPTION
**Pull request by tgross**
Implementation of the `CreateVolume`, `DeleteVolume`, and `ListExternalVolume` HTTP endpoints for CSI.

Notes for reviewers:
* Note there are no CLI commits; those will be handled in the next PR but I've got https://github.com/hashicorp/nomad/pull/10211 that uses this API so I feel comfortable with the "shape".
* HTTP API docs will be in a separate PR
* This PR is targeting the `csi-create-volume` branch: https://github.com/hashicorp/nomad/pull/10165, which has the RPCs already merged in.

### CODE REVIEW

**Review by cgbaker — APPROVED** [review]
LGTM 🚀

File: command/agent/csi_endpoint.go [hunk 1]
```diff
@@ -142,6 +181,33 @@ func (s *HTTPServer) csiVolumePut(resp http.ResponseWriter, req *http.Request) (
...
+	s.parseWriteRequest(req, &args.WriteRequest)
+
+	var out structs.CSIVolumeCreateResponse
+	if err := s.agent.RPC("CSIVolume.Create", &args, &out); err != nil {
+		return nil, err
+	}
+
+	setMeta(resp, &out.QueryMeta)
+
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (tgross) [reply command/agent/csi_endpoint.go:208]: Good catch... we should be handing back the external volume ID at least.

This comment also made me have some thoughts about idempotency of the volume create call but that's out of scope for this PR (the RPCs where this is controlled are merged to `csi-volume-create`). After I do some testing I'll open a PR back to that branch if we need to fix anything there.

File: command/agent/csi_endpoint.go [hunk 2]
```diff
@@ -142,6 +181,33 @@ func (s *HTTPServer) csiVolumePut(resp http.ResponseWriter, req *http.Request) (
...
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	args0 := structs.CSIVolumeCreateRequest{}
+	if err := decodeBody(req, &args0); err != nil {
+		return err, CodedError(400, err.Error())
+	}
+
+	args := structs.CSIVolumeCreateRequest{
```
Comment (tgross) [inline command/agent/csi_endpoint.go:196]: Whoops, not sure what happened here either. Need to fix this.

File: command/agent/csi_endpoint.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgross) [reply command/agent/csi_endpoint.go:208]: Done in aa2264b

**Pull request merged**